hpx/parallel/algorithms/fill.hpp

```diff
@@ -65,10 +62,10 @@ namespace hpx { namespace parallel { inline namespace v1
               : fill::algorithm("fill")
             {}
 
-            template <typename ExPolicy, typename InIter, typename T>
+            template <typename ExPolicy, typename FwdIter, typename T>
             HPX_HOST_DEVICE
             static hpx::util::unused_type
-            sequential(ExPolicy, InIter first, InIter last,
+            sequential(ExPolicy, FwdIter first, FwdIter last,
```

Comment:
Right, it will never be called with an input iterator. But, regardless of execution flow, as one independent function, clarifying iterator requirements seems better. If anyone refer this function for another implementation, he(she) may misunderstand that this function only works with forward iterator.